src/tools/language-server/language-service.ts

```diff
@@ -8,90 +8,105 @@
+    this.connection.onExit(this.exit.bind(this));
+
+    // Register language service handlers
+
+    // TODO: this.connection.onCompletion(this.completion);
+    this.connection.onDidSaveTextDocument(this.didSaveTextDocument.bind(this));
+    // TODO: this.connection.onDidChangeConfiguration(this.didChangeConfig.bind(this));
+    // TODO: this.connection.onDidChangeTextDocument(this.didChange.bind(this));
+    this.connection.onExecuteCommand(this.executeCommand.bind(this));
 
-    this.reader.onError(err => {
-      this.context.logger.error(err);
-    });
+    this.connection.listen();
   }
 
-  async update(message: Message): Promise<Message | undefined> {
-    const logger = this.context.logger;
-    // Ignore responses (currently unhandled).
-    if (isResponseMessage(message)) {
-      logger.info('Received response message:', message);
-      return undefined;
-    }
-    if (!isRequestMessage(message) && !isNotificationMessage(message)) {
-      logger.error('Received invalid message:', message);
-      return undefined;
-    }
-    logger.info('Received valid message:', message);
-
-    const method = camelCase(message.method);
-    switch (method) {
-      case 'initialize':
-        this.initialized = true;
-        this.streaming = message.params.capabilities.streaming;
-        break;
-      case 'shutdown':
-        this.initialized = false; // TODO(cypher1): Cleanup.
-        // Ignore as the service will close when the socket does.
-        return undefined;
-      case 'exit':
-        // Ignore as the service will close when the socket does.
-        return undefined;
-      default: // Fall through to use handlers[method].
-    }
-    // The message needs to be handler by an appropriate handler.
-    const handler = handlers[method];
-
-    if (typeof handler !== 'function') {
-      // Method not implemented
-      if (isRequestMessage(message)) {
-        return {
-          jsonrpc,
-          id: message.id,
-          error: {
-            code: ErrorCodes.MethodNotFound,
-            message: `Method ${method} not implemented`,
-          },
-        } as Message;
-      } else {
-        logger.warn(`Method ${method} not implemented`);
+  // Handlers
+
+  initialize(params: InitializeParams): InitializeResult {
+    this.rootPath = params.rootPath || uri2path(params.rootUri);
+
+    return {
+      capabilities: {
+        // Tell the client that the server works in FULL text document sync mode
+        textDocumentSync: TextDocumentSyncKind.Full,
+        /*
+           hoverProvider: false,
+           signatureHelpProvider: {
+           triggerCharacters: ['(', ','],
+           },
+           definitionProvider: false, // TODO(cypher1): Provide definitions.
+           typeDefinitionProvider: false,
+           referencesProvider: false,
+           documentSymbolProvider: false,
+           workspaceSymbolProvider: false,
+           xdefinitionProvider: false,
+         */
```

Comment:
How come all this is commented out? Just delete it?

Reply:
I couldn't think of a better way to keep track of planned features than in the list of features.
Done